src/shared/services/rank-storage-service.ts

```
@@ -0,0 +1,90 @@
+                            } else {
+                                addedResult.push({ added: false, item: { hostname: newRank.hostname, keyWords: [targetWithKeyword] } });
+                            }
+                        } else {
+                            targetSite.keyWords.push({
+                                keyword: newRank.keyWords[0].keyword,
+                                rankHistory: [...newRank.keyWords[0].rankHistory]
+                            });
+                            addedResult.push({ added: true, item: { hostname: newRank.hostname, keyWords: [{ keyword: newRank.keyWords[0].keyword, rankHistory: newRank.keyWords[0].rankHistory }] } });
+                        }
+                    }
+                });
+                chrome.storage.sync.set({ 'myranks': updatedRanks }, () => {
+                    resolve(addedResult);
+                });
+            });
+        });
+    }
+    public removeRanks(name: string): void {
+        chrome.storage.sync.get('myranks', (data) => {
+            const mySites: RankStorageModel[] = data.myranks || [];
+
+            var targetSite = mySites.find(site => site.hostname === name);
+            if (targetSite) {
+                var index = data.myranks.indexOf(targetSite);
+                data.myranks.splice(index, 1);
+            }
+            chrome.storage.sync.set({ 'myranks': data.myranks });
+        })
+    }
+    public findRanks(resultItems: Element[], keyword: any): Promise<RankStorageModel[]> {
+        return new Promise((resolve, reject) => {
+            var rankStorageList: RankStorageModel[] = [];
+            chrome.storage.sync.get('mysites', (data) => {
+                if (data.mysites && data.mysites.length > 0) {
+                    const mysites: SiteStorageModel[] = data.mysites;
+                    const today = new Date().toISOString().split("T")[0];
+                    // Throw an error if the keyword is null or undefined
```

Why does it throw an error in case of a non-existent keyword? Can't it just return? And additionally, the error is not handled in its usage.